### PR TITLE
refactor(service-client): Pass the correct amqp audience based on aut…

### PR DESCRIPTION
**Pull request by vinagesh**
…h type

## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

### CODE REVIEW

File: iothub/service/src/IotHubConnection.cs [hunk 1]
```diff
@@ -553,7 +553,7 @@ private async Task SendCbsTokenAsync(AmqpCbsLink cbsLink, TimeSpan timeout)
                     Credential.AmqpEndpoint,
                     audience,
                     resource,
-                    AccessRightsHelper.AccessRightsToStringArray(_accessRights),
+                    Credential.AmqpAudience,
```
Comment (vinagesh) [inline iothub/service/src/IotHubConnection.cs:556]: The IoT hub service ignores this value for AAD auth. But for the sake of correctness, we should not be passing audience related to SAS tokens during AAD auth.

**Review by timtay-microsoft — APPROVED** [review]

File: iothub/service/src/IotHubConnectionProperties.cs [hunk 2]
```diff
@@ -35,6 +35,12 @@ protected IotHubConnectionProperties(string hostName)
             IotHubName = GetIotHubName(hostName);
             AmqpEndpoint = new UriBuilder(CommonConstants.AmqpsScheme, HostName, AmqpConstants.DefaultSecurePort).Uri;
             HttpsEndpoint = new UriBuilder(HttpsEndpointPrefix, HostName).Uri;
+#if NET451
+            AmqpAudience = new string[] { };
+
```
Comment (drwill-ms) [inline iothub/service/src/IotHubConnectionProperties.cs:40]: extra newline here

**Review by drwill-ms — APPROVED** [review]
Are you j

File: iothub/service/src/IotHubConnectionProperties.cs (hunk 2 above)
Comment (vinagesh) [reply iothub/service/src/IotHubConnectionProperties.cs]: I ended up making this code better by using lists :) I'm not a fan of these #if statements.

**Pull request merged**